docs/milestones/2.md

```diff
@@ -0,0 +1,89 @@
+# Milestone 2
+
+The second milestone builds on Milestone 1 by including a single slot proof voting strategy, an L1 execution strategy, and finally an L1 interaction [1] contract that we call StarkNet Commit along with a corresponding auth contract for this type of interaction on L2. Additionally, we update the space contract to take an array of voting and auth strategies rather than just a single option for each. 
+
+![](./architecture.png)
+
+[Source](https://whimsical.com/snapshot-x-milestone-2-7UJARrFFC3LbnZexGEtQQB)
+
+### Single Slot Proof Voting Strategy Contract
+
+We add the principal voting strategy of Snapshot X to this milestone. The single slot proof strategy will allow the balances of the majority [2] of ERC20 and ERC721 tokens to be used to find voting power. 
+
+#### Functions
+
+- **get_voting_power(address: EthAddress, ethereum_block_number: felt, params: felt array)**  
+Returns the voting power for a specific address at a specific Ethereum block number **block_number**, **params** are the storage proof data required by the Fossil module to prove the contents of a slot. In the case of finding the voting power from an ERC20 balance, the slot will correspond to the storage slot in the ERC20 contract where the balance of **address** is stored. 
+
+### StarkNet Commit Contract
```

Comment:
`Starknet Commit Contract` sounds like it something "bigger" than Snapshot X. As if anyone on Starknet could benefit from it?

Reply:
yeah me and fabien discussed this on discord, could be used by anyone who wants to commit data to L2 via an L1 tx.